### DISCUSSION TITLE
Any consideration given to working with Planet CNC cncusb devices?

### DISCUSSION THREAD

**Discussion by VA1DER** [winder/Universal-G-Code-Sender#2271] [General]
Has there been any consideration given in the past to working with Planet CNC's "cncusb" devices?  Their hardware is nice, but software for it is basically a monoculture.  I'd love to see ugcs work with it.

**Comment by breiler** [comment]
I think their software (PlanetCNC) looks really neat and polished and it has some really nice features. 

The protocol for communicating with their controller is not public so that would require us to reverse engineer it. That would be quite an investment of both time and money...

I think that it is better to focus on improving support for the existing open source controllers.

**Comment by VA1DER** [comment]
> I think their software (PlanetCNC) looks really neat and polished and it has some really nice features.

It's not bad, actually, but it's not open source.

> The protocol for communicating with their controller is not public so that would require us to reverse engineer it.

The controller is responding to straight g-code.

> I think that it is better to focus on improving support for the existing open source controllers.

The original controller was released as an [open source hardware](http://web.archive.org/web/20120105151838/http://www.planet-cnc.com/diy/schematicsV10.pdf) project that uses a PIC18F family microcontroller.  In reality, the design is little more than a few very minor components (pull up resistors and filter capacitors) to interface the outputs of a PIC18F4550 to the outside world.  It receives g-code over the 4550's USB-Serial connection and sent out stepper signals over the controller's digital IO pins.  It was originally intended to interface with open source [PicStep](https://github.com/alangarf/picstep) stepper controllers.

It would be nice to be able to get back to the design's open source hardware roots and integrate it with open source software.  I am very interested in pursuing this.  If I am able to get it to talk to UGS, would you be willing to incorporate the patches?
